classes/Combination.php

```diff
@@ -94,6 +101,10 @@ class CombinationCore extends ObjectModel
             'low_stock_alert' => ['type' => self::TYPE_BOOL, 'shop' => true, 'validate' => 'isBool'],
             'default_on' => ['type' => self::TYPE_BOOL, 'allow_null' => true, 'shop' => true, 'validate' => 'isBool'],
             'available_date' => ['type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDateFormat'],
+
+            /* Lang fields */
+            'available_now' => ['type' => self::TYPE_STRING, 'lang' => true, 'validate' => 'isGenericName', 'size' => 255],
+            'available_later' => ['type' => self::TYPE_STRING, 'lang' => true, 'validate' => 'IsGenericName', 'size' => 255],
```

Comment:
It would be nice to define these "size" values in dedicated "CombinationSettings" class. (see example [ProductSettings](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Product/ProductSettings.php)). Then they can be reused in backOffice form types for constraints and if needed anywhere else (it would be the source of truth).

Basically these "{Foo}Settings" classes should be created for all such constants if there is no related ValueObject (if there is a value object, then the related constant stays there). Its kind of prestashop convention but idk if its written anywhere :smile: We have some examples with other domain classes - like SupplierSettings, AddressSettings (but probably we didn't went that far to also use those constants in related object model, altough we should have)

Reply:
done